web/scm/src/app/side-panel/side-panel.component.ts

```diff
@@ -1,15 +1,95 @@
 import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormControl } from '@angular/forms';
 
 @Component({
   selector: 'scm-side-panel',
   templateUrl: './side-panel.component.html',
   styleUrls: ['./side-panel.component.scss']
 })
 export class SidePanelComponent implements OnInit {
+  form: FormGroup;
 
-  constructor() { }
+  productSelectOptions: any[];
+  categorySelectOptions: any[];
+  supplierSelectOptions: any[];
+
+  countrySelectOptions: any[];
+  regionSelectOptions: any[];
+  citySelectOptions: any[];
+
+  constructor() {
+    this.form = new FormGroup({
+      productFilterGroup: new FormGroup({
+        productSelect: new FormControl(),
+        categorySelect: new FormControl(),
+        supplierSelect: new FormControl()
+      }),
+      locationFilterGroup: new FormGroup({
+        countrySelect: new FormControl(),
+        regionSelect: new FormControl(),
+        citySelect: new FormControl(),
+      }),
+      additionalFilterGroup: new FormGroup({
+        minLeadTimeInput: new FormControl(),
+        maxLeadTimeInput: new FormControl()
+      }),
+      mapPropFilterGroup: new FormGroup({
+        uniqueCheckbox: new FormControl(),
+        drawPathCheckbox: new FormControl(),
+        singleSrcCheckbox: new FormControl(),
+      })
+    });
+  }
 
   ngOnInit(): void {
+    this.initializeOptions();
+  }
+
+  initializeOptions() {
```

Comment:
OMG, could not found codestyle reference, but for dart and most of java there is rule -> if it's public, please add doc. Simple doc describing what function does

Reply:
Yeah I looked it up too, no standard for angular. I will just use JSDoc style.